python/foglamp/plugins/south/sinusoid/sinusoid.py

```diff
@@ -166,35 +168,40 @@ def generate_data():
 
     async def save_data():
         try:
+            time_time = time.time
+            start = time_time()
+            recs = int(handle['dataPointsPerSec']['value'])
+            period = round(1.0 / recs, len(str(recs))+1)
```

Comment:
why the ZeroDivisionError code is not applied here as in the deleted code.

Reply:
restored